lib/cms/behaviors/publishing.rb

```diff
@@ -130,28 +130,28 @@ def publish!
 
                   d.update_attributes(:published => true)
 
-                  # copy values from the draft to the main record
-                  quoted_attributes = d.send(:arel_attributes_with_values_for_update, self.class.versioned_columns)
 
                   #the values from the draft MAY have a relation of the versioned module
                   #as opposed to the actual class itself
                   #eg Page::Version, and not Page
-                  #so remap to the actual arel_table´
+                  #so remap to the actual table´
                   #I haven't figured out why this is, but I know it happens when you call save! on Page
                   #during seeding of data
-                  if self.class.arel_table.name != quoted_attributes.keys[0].relation.name
-                    quoted_attributes = quoted_attributes.inject({}) { |hash, pair| hash[self.class.arel_table[pair[0].name]] = pair[1]; hash }
+                  my_class = self.class.name.split("::Version").first.constantize
+                  versioned_columns = my_class.versioned_columns
+                  update_params = {}
+                  versioned_columns.each do |c|
+                    update_params[c.to_sym] = d.send(c)
                   end
 
-                  # Doing the SQL ourselves to avoid callbacks
-                  ActiveRecord::Base.connection.execute(self.class.unscoped.where(self.class.arel_table[self.class.primary_key].eq(id)).arel.compile_update(quoted_attributes, id).to_sql)
+                  my_class.where(:id=>id).update_all(update_params)
```

Comment:
This was causing lots of deprecation warnings so I moved it to active record.  The update_all should avoid callbacks on the model.